### DISCUSSION TITLE
Spring Security and Oauth2

### DISCUSSION THREAD

**Discussion by morrowyn** [ExpediaGroup/graphql-kotlin#896] [Q&A] [answered]
Hi, 

Is it possible to secure the graphql subscriptions with spring-security 5.x and oauth2/jwt? 

Regards,

**Comment by dariuszkuc** [comment, marked as answer]
According to [Spring documentation](https://docs.spring.io/spring-security/site/docs/current/reference/html5/#websocket) yes you should be able to use `spring-security` with GraphQL Subscriptions (which are WebSocket based).

See https://github.com/ExpediaGroup/graphql-kotlin/discussions/663 for example security web filter.

**Reply by morrowyn** [reply]
Thank you. I will look into this.
